src/plugins/Transloadit/index.js

```diff
@@ -0,0 +1,111 @@
+const Plugin = require('../Plugin')
+const Tus10Plugin = require('../Tus10')
+const Client = require('./Client')
+
+module.exports = class Transloadit extends Plugin {
+  constructor (core, opts) {
+    super(core, opts)
+    this.type = 'uploader'
+    this.id = 'Transloadit'
+    this.title = 'Transloadit'
+
+    const defaultOptions = {
+      templateId: null,
+      resume: true,
+      allowPause: true
+    }
+
+    this.opts = Object.assign({}, defaultOptions, opts)
+
+    if (!this.opts.key) {
+      throw new Error('Transloadit: The `key` option is required. ' +
+        'You can find your Transloadit API key at https://transloadit.com/accounts/credentials.')
+    }
+    if (!this.opts.templateId) {
+      throw new Error('Transloadit: The `templateId` option is required. ' +
+        'You can find your template\'s ID at https://transloadit.com/templates.')
+    }
+
+    this.client = new Client({
+      key: this.opts.key
+    })
+  }
+
+  createAssembly () {
+    this.core.log('Transloadit: create assembly')
+
+    return this.client.createAssembly({
+      templateId: this.opts.templateId,
+      expectedFiles: Object.keys(this.core.state.files).length
+    }).then((assembly) => {
+      this.updateState({ assembly })
+
+      function attachAssemblyMetadata (file, assembly) {
+        const meta = Object.assign({}, file.meta, {
+          assembly_url: assembly.assembly_url,
+          filename: file.name,
+          fieldname: 'file'
+        })
```

Comment:
The Tus plugin uses the `meta` property as an `Upload-Meta` header. (should add a code comment here explaining that)